lib/pdu.c

```diff
@@ -397,15 +397,23 @@ smb2_find_pdu(struct smb2_context *smb2,
 static int
 smb2_is_error_response(struct smb2_context *smb2,
                        struct smb2_pdu *pdu) {
-        if ((smb2->hdr.status & SMB2_STATUS_SEVERITY_MASK) ==
+        if ((smb2->hdr.status & SMB2_STATUS_ERROR_SEVERITY_MASK) ==
             SMB2_STATUS_SEVERITY_ERROR) {
                 switch (smb2->hdr.status) {
                 case SMB2_STATUS_MORE_PROCESSING_REQUIRED:
                         return 0;
                 default:
                         return 1;
                 }
-        }
+        } else if ((smb2->hdr.status & SMB2_STATUS_WARNING_SEVERITY_MASK) ==
```

Comment:
Aside from the changes to the MASKS, remove these changes, otherwise the patch looks fine.
Please update it and I will merge it.

Thanks!